src/store/src/Bridge/Qdrant/Store.php

```diff
@@ -0,0 +1,145 @@
+{
+    public function __construct(
+        private HttpClientInterface $httpClient,
+        private string $endpointUrl,
+        #[\SensitiveParameter] private string $apiKey,
+        private string $collectionName,
+        private int $embeddingsDimension = 1536,
+        private string $embeddingsDistance = 'Cosine',
+    ) {
+    }
+
+    public function add(VectorDocument ...$documents): void
+    {
+        $this->request('PUT', \sprintf('collections/%s/points', $this->collectionName), [
+            'points' => array_map($this->convertToIndexableArray(...), $documents),
+        ]);
+    }
+
+    /**
+     * @param array{
+     *     limit?: positive-int,
+     *     offset?: positive-int
+     * } $options
+     */
+    public function query(Vector $vector, array $options = [], ?float $minScore = null): array
+    {
+        $payload = [
+            'vector' => $vector->getData(),
+            'with_payload' => true,
+            'with_vector' => true,
+        ];
+
+        if (\array_key_exists('limit', $options)) {
+            $payload['limit'] = $options['limit'];
+        }
+
+        if (\array_key_exists('offset', $options)) {
+            $payload['offset'] = $options['offset'];
+        }
+
+        $response = $this->request('POST', \sprintf('collections/%s/points/query', $this->collectionName), $payload);
+
+        return array_map($this->convertToVectorDocument(...), $response['result']['points']);
+    }
+
+    public function initialize(array $options = []): void
+    {
+        if ([] !== $options) {
+            throw new InvalidArgumentException('No supported options');
+        }
+
+        $collectionExistResponse = $this->request('GET', \sprintf('collections/%s/exists', $this->collectionName));
+
+        if ($collectionExistResponse['result']['exists']) {
+            return;
+        }
+
+        $this->request('PUT', \sprintf('collections/%s', $this->collectionName), [
+            'vectors' => [
+                'size' => $this->embeddingsDimension,
+                'distance' => $this->embeddingsDistance,
+            ],
+        ]);
+    }
+
+    /**
+     * @param array<string, mixed> $payload
+     *
+     * @return array<string, mixed>
+     */
+    private function request(string $method, string $endpoint, array $payload = []): array
+    {
+        $url = \sprintf('%s/%s', $this->endpointUrl, $endpoint);
+
+        $response = $this->httpClient->request($method, $url, [
+            'headers' => [
+                'api-key' => $this->apiKey,
+            ],
+            'json' => $payload,
+        ]);
+
+        return $response->toArray();
+    }
+
+    /**
+     * @return array<string, mixed>
+     */
+    private function convertToIndexableArray(VectorDocument $document): array
+    {
+        return array_merge([
+            'id' => $document->id->toRfc4122(),
+            'vector' => $document->vector->getData(),
+        ], $document->metadata->getArrayCopy());
```

Comment:
In the end we need the metadata to be looped through the store, if that's not possible with the engine, we would need to use the `id` to hydrate the record from another store. Luckily Qdrant has the concept of `payload`, that we can use here:
```suggestion
        return [
            'id' => $document->id->toRfc4122(),
            'vector' => $document->vector->getData(),
            'payload' => $document->metadata->getArrayCopy(),
        ];
```
See https://api.qdrant.tech/api-reference/points/upsert-points

Reply:
Ah, clever, I saw the `payload` option but thought it was related to the "content" of the document.